src/retro_data_structures/disc/wii_disc.py

```diff
@@ -0,0 +1,344 @@
+from __future__ import annotations
+
+import enum
+import typing
+
+import construct
+from Crypto.Cipher import AES
+
+from retro_data_structures.adapters.enum_adapter import EnumAdapter
+from retro_data_structures.disc import disc_common
+
+if typing.TYPE_CHECKING:
+    from pathlib import Path
+
+COMMON_KEYS = [
+    # Normal
+    b"\xeb\xe4\x2a\x22\x5e\x85\x93\xe4\x48\xd9\xc5\x45\x73\x81\xaa\xf7",
+    # Korean
+    b"\x63\xb8\x2b\xb4\xf4\x61\x4e\x2e\x13\xf2\xfe\xfb\xba\x4c\x9b\x7e",
+]
+
+
+class ShiftedInteger(construct.Adapter):
+    def _decode(self, obj, context, path):
+        return obj << 2
+
+    def _encode(self, obj, context, path):
+        return obj >> 2
+
+
+WiiDiscHeader = construct.Struct(
+    game_code=construct.Bytes(4),
+    maker_code=construct.Bytes(2),
+    disc_id=construct.Int8ub,  # for multi-disc games
+    version=construct.Int8ub,
+    audio_streaming=construct.Int8ub,
+    stream_buffer_size=construct.Int8ub,
+    _unused_a=construct.Const(b"\x00" * 14),
+    _wii_magic_word=construct.Const(0x5D1C9EA3, construct.Int32ub),
+    _gc_magic_word=construct.Const(0, construct.Int32ub),
+    game_name=construct.PaddedString(64, "utf8"),
+    disable_hash_verification=construct.Flag,
+    disable_disc_encryption=construct.Flag,
+    _unused_b=construct.Const(b"\x00" * 0x39E),
+    debug_monitor_offset=construct.Int32ub,
+    debug_monitor_load_address=construct.Int32ub,
+    _unused_c=construct.Const(b"\x00" * 24),
+    main_executable_offset=ShiftedInteger(construct.Int32ub),
+    fst_offset=ShiftedInteger(construct.Int32ub),
+    fst_size=ShiftedInteger(construct.Int32ub),
```

Comment:
are the only differences between GC and Wii headers the magic words and the shifted integers? surely the constructs could be combined instead of duplicated like this

Reply:
I swapped between using the same, then not the same. The main thing was shifted integers though.

Reply:
Any suggestions on how to automatically handle the shifted integers in a way that isn't overly annoying?

Reply:
`PossiblyShiftedInteger = IfThenElse(lambda ctx: ctx.is_wii, ShiftedInteger(Int32ub), Int32ub)`

and add a `Computed` field for `is_wii` after the magic words?